py3status/modules/process_status.py

```diff
@@ -3,66 +3,81 @@
+    icon_off: display when the process is not running (default '■')
+    icon_on: display when the process is running (default '●')
     process: the process name to check if it is running (default None)
+    string_unavailable: no process name (default 'process_status: N/A')
+
+Format placeholders:
+    {icon} process icon
+    {process} process name
 
 Color options:
-    color_bad: Process not running or error
-    color_good: Process running
+    color_bad: process currently not running or unavailable
+    color_good: process currently running
 
 @author obb, Moritz Lüdecke
 """
 
-import os
-import subprocess
-
-ERR_NO_PROCESS = 'no process name given'
-
 
 class Py3status:
     """
     """
     # available configuration parameters
     cache_timeout = 10
-    format_not_running = u'■'
-    format_running = u'●'
+    format = '{icon}'
     full = False
+    icon_off = u'■'
+    icon_on = u'●'
     process = None
+    string_unavailable = 'process_status: N/A'
+
+    class Meta:
+        deprecated = {
+            'rename': [
+                {
+                    'param': 'format_running',
+                    'new': 'icon_on',
+                    'msg': 'obsolete parameter use `icon_on`',
+                },
+                {
+                    'param': 'format_not_running',
+                    'new': 'icon_off',
+                    'msg': 'obsolete parameter use `icon_off`',
+                },
+            ],
+        }
 
-    def _get_text(self):
-        fnull = open(os.devnull, 'w')
+    def _is_running(self):
         pgrep = ["pgrep", self.process]
 
         if self.full:
             pgrep = ["pgrep", "-f", self.process]
-
-        if subprocess.call(pgrep,
-                           stdout=fnull, stderr=fnull) == 0:
-            text = self.format_running
-            color = self.py3.COLOR_GOOD
+        try:
+            self.py3.command_output(pgrep)
+        except:
+            return False
         else:
-            text = self.format_not_running
-            color = self.py3.COLOR_BAD
-
-        return (color, text)
+            return True
 
     def process_status(self):
+        response = {'cached_until': self.py3.time_in(self.cache_timeout)}
+
         if self.process is None:
-            color = self.py3.COLOR_BAD
-            text = ERR_NO_PROCESS
+            response['cached_until'] = self.py3.CACHE_FOREVER
+            response['color'] = self.py3.COLOR_BAD
+            response['full_text'] = self.string_unavailable
         else:
-            (color, text) = self._get_text()
+            run = self._is_running()
+            icon = self.icon_on if run else self.icon_off
 
-        response = {
-            'cached_until': self.py3.time_in(self.cache_timeout),
-            'full_text': text,
-            'color': color
-        }
+            response['color'] = self.py3.COLOR_ON or self.py3.COLOR_GOOD \
+                if run else self.py3.COLOR_OFF or self.py3.COLOR_BAD
```

Comment:
The thing is. 

1) color_good/bad are defined by default whereas color_on/off are not unless the user has set them.

2) color_on/off have been added by you in this PR so you are changing the behaviour.  If I don't have them set then I'd lose color clues.

now I think having color_on/off is no bad thing as being on/off is different to good/bad as you said somewhere. but we should default them with good/bad.

If I was doing this now I might do
```
def post_config_hook(self):
    self.color_on = self.py3.COLOR_ON or self.py3.COLOR_GOOD
    self.color_off = self.py3.COLOR_OFF or self.py3.COLOR_BAD

...

response['color'] = self.color_on if run else self.color_off
```
as it looks a bit more readable.

but the bottom line is yes you need both because otherwise you are changing behaviour for users rather than giving an enhancement.